src/vm/assembly.cpp

```diff
@@ -2583,14 +2586,21 @@ HRESULT RunMain(MethodDesc *pFD ,
 
         if (pParam->pFD->IsVoid()) 
         {
-            // Set the return value to 0 instead of returning random junk
-            *pParam->piRetVal = 0;
             threadStart.Call(&stackVar);
+            *pParam->piRetVal =
+#ifndef FEATURE_CORECLR
+                // Set the return value to 0 instead of returning random junk
+                0;
```

Comment:
Since CoreCLR is always hosted, it seems that based on the behavior of the desktop, I think that we could remove the modification of the latched code here completely for FEATURE_CORECLR. That would mean ifdef out the two calls to SetLatchedExitCode (one with zero, the other with the return value of the main).
Otherwise it seems ok to me.